ygen/protogen.go

```diff
@@ -668,12 +824,18 @@ func protoListDefinition(args protoDefinitionArgs) (*protoMsgListField, *protoMs
 	var listDef *protoMsgListField
 	if !isKeyedList(listMsg.entry) {
 		// In proto3 we represent unkeyed lists as a
-		// repeated field of the parent message.
-		p := fmt.Sprintf("%s.%s.%s", args.basePackageName, childPkg, listMsgName)
-		p, _ = stripPackagePrefix(fmt.Sprintf("%s.%s", args.basePackageName, args.parentPackage), p)
-		listDef = &protoMsgListField{
-			listType: p,
-			imports:  []string{importPath(args.baseImportPath, args.basePackageName, childPkg)},
+		// repeated field of the list message.
+		if !args.nestedMessages {
+			p := fmt.Sprintf("%s.%s.%s", args.basePackageName, childPkg, listMsgName)
+			p, _ = stripPackagePrefix(fmt.Sprintf("%s.%s", args.basePackageName, args.parentPackage), p)
+			listDef = &protoMsgListField{
+				listType: p,
+			}
+			listDef.imports = []string{importPath(args.baseImportPath, args.basePackageName, childPkg)}
+		} else {
```

Comment:
Consider removing else and putting this as default init.

Reply:
Done.